server/tests/db/test_01_es5.js

```diff
@@ -0,0 +1,1739 @@
+};
+
+tests.xxx = function() {
+  var a = 56;
+  while (false) {
+    a++;
+  }
+  console.assert(a === 56, 'whileFalse');
+};
+
+tests.doWhileFalse = function() {
+  var a = 56;
+  do {
+    a++;
+  } while (false);
+  console.assert(a === 57, 'doWhileFalse');
+};
+
+tests.breakDoWhile = function() {
+  var a = 57;
+  do {
+    a++;
+    break;
+    a++;
+  } while (false);
+  console.assert(a === 58, 'breakDoWhile');
+};
+
+tests.selfBreak = function() {
+  console.assert(eval('foo: break foo;') === undefined, 'selfBreak');
+};
+
+tests.breakWithFinally = function() {
+  var a = 6;
+  foo: {
+    try {
+      a *= 10;
+      break foo;
+    } finally {
+      a--;
+    }
+  }
+  console.assert(a === 59, 'breakWithFinally');
+};
+
+tests.continueWithFinally = function() {
+  var a = 59;
+  do {
+    try {
+      continue;
+    } finally {
+      a++;
+    }
+  } while (false);
+  console.assert(a === 60, 'continueWithFinally');
+};
+
+tests.breakWithFinallyContinue = function() {
+  var a = 0;
+  while (a++ < 60) {
+    try {
+      break;
+    } finally {
+      continue;
+    }
+  }
+  console.assert(a === 61, 'breakWithFinallyContinue');
+};
+
+tests.returnWithFinallyContinue = function() {
+  var f = function() {
+    var i = 0;
+    while (i++ < 61) {
+      try {
+        return 42;
+      } finally {
+        continue;
+      }
+    }
+    return i;
+  };
+  console.assert(f() === 62, 'returnWithFinallyContinue');
+};
+
+tests.or = function() {
+  console.assert((63 || 'foo') === 63, 'orTrue');
+  console.assert((false || 64) === 64, 'orFalse');
+  var r = 0;
+  true || (r++);
+  console.assert(r === 0, 'orShortcircuit');
+};
+
+tests.and = function() {
+  console.assert((({}) && 65) === 65, 'andTrue');
+  console.assert((0 && 65) === 0, 'andFalse');
+  var r = 0;
+  false && (r++);
+  console.assert(r === 0, 'andShortcircuit');
+};
+
+tests.forTriangular = function() {
+  var t = 0;
+  for (var i = 0; i < 12; i++) {
+    t += i;
+  }
+  console.assert(t === 66, 'forTriangular');
+};
+
+tests.forIn = function() {
+  var x = 0, a = {a: 60, b:3, c:4};
+  for (var i in a) { x += a[i]; }
+  console.assert(x === 67, 'forIn');
+};
+
+tests.forInMemberExp = function() {
+  var x = 1, o = {foo: 'bar'}, a = {a:2, b:2, c:17};
+  for (o.foo in a) { x *= a[o.foo]; }
+  console.assert(x === 68, 'forInMemberExp');
+};
+
+tests.forInMembFunc = function() {
+  var x = 0, o = {};
+  var f = function() { x += 20; return o; };
+  var a = {a:2, b:3, c:4};
+  for (f().foo in a) { x += a[o.foo]; }
+  console.assert(x === 69, 'forInMembFunc');
+};
+
+tests.methodCall = function() {
+  var o = {
+    f: function() { return this.foo; },
+    foo: 70
+  };
+  console.assert(o.f() === 70, 'methodCall');
+};
+
+tests.demethodedCall = function() {
+  var o = { f: function() { return this; }};
+  var g = o.f;
+  console.assert(g() === undefined, 'demethodedCall');
+};
+
+tests.testThis = function() {
+  console.assert(this === tests, 'testThis');
```

Comment:
This is not the same as the original test, which tests the value of `this` at the top level (i.e., that it is `undefined` rather an alias of `window` or the like).